### PR TITLE
feat/add button loading spinner

### LINKED ISSUE #83: Button component - Handle loading state
**Story**
Button component should take `loading` prop and render a simple animated spinner when `loading={true}`

**Tasks**
- [x] add simple loading spinner component with CSS
- [x] add `loading` prop to **Button** component and set the initial value to false

### PR DESCRIPTION
**Pull request by atwfk**
resolve #83

### CODE REVIEW

**Comment by AhmedAdel999** [comment]
Don't you think it would be better to offer the option to show the loading spinner before or after the text?

File: src/components/Button/Button.stories.mdx [hunk 1]
```diff
@@ -39,6 +39,14 @@ Users can change color of the button, by passing `color` prop.
   <Story name="Neutral" args={{color: "neutral", children: "Neutral"}}>{Template.bind({})}</Story>
 </Canvas>
 
+## Loaing
+
+set `loading` to true to show animated spinner instead of the text
```
Comment (AhmedAdel999) [inline src/components/Button/Button.stories.mdx:44]: There is a typo in `loaing`

File: src/components/Button/Button.module.css [hunk 2]
```diff
@@ -81,3 +81,16 @@
 .button-s {
   @apply font-medium leading-20 text-14 tracking-wide;
 }
+
+/* Loading spinner styles */
+.loading-wrapper {
+  @apply absolute -translate-x-1/2 -translate-y-1/2 top-1/2 left-1/2 inline-block;
+}
+
+.loading {
```
Comment (AhmedAdel999) [inline src/components/Button/Button.module.css:90]: I would suggest adding `cursor-wait` class.

Comment (atwfk) [reply src/components/Button/Button.module.css:90]: I guess it's better to add `cursor-not-allowed` & also I shouldn't leave it clickable in loading state, it should be disabled,
I will update it RN

**Comment by AhmedAdel999** [comment]
[https://mui.com/material-ui/react-button/#loading-button](https://mui.com/material-ui/react-button/#loading-button )take a look here.

**Comment by atwfk** [comment]
> https://mui.com/material-ui/react-button/#loading-buttontake a look here.

hmmm can we stick with just loading the spinner and hide the text for now? and we can add more cases in the future

**Pull request merged**